mypy/semanal.py

```diff
@@ -2906,6 +2909,16 @@ def visit_overloaded_func_def(self, func: OverloadedFuncDef) -> None:
         func._fullname = self.sem.qualified_name(func.name())
         if kind == GDEF:
             self.sem.globals[func.name()] = SymbolTableNode(kind, func, self.sem.cur_mod_id)
+        if func.impl:
+            # Also analyze the function body (in case there are conditional imports).
```

Comment:
Hmm... Refactor this to merge with the nearly identical block in the previous method?